config/load.go

```diff
@@ -0,0 +1,210 @@
+package config
+
+import (
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/fsnotify/fsnotify"
+	"github.com/joho/godotenv"
+	"github.com/spf13/viper"
+)
+
+func (c *Config) load() {
+	c.hotReloadableConfig = make(map[string][]*configValue)
+	c.nonHotReloadableConfig = make(map[string][]*configValue)
+	c.envs = make(map[string]string)
+
+	if err := godotenv.Load(); err != nil {
+		fmt.Println("INFO: No .env file found.")
+	}
+
+	configPath := getEnv("CONFIG_PATH", "./config/config.yaml")
+
+	v := viper.NewWithOptions(viper.EnvKeyReplacer(&envReplacer{c: c}))
+	v.AutomaticEnv()
+	bindLegacyEnv(v)
+
+	v.SetConfigFile(configPath)
+	err := v.ReadInConfig() // Find and read the config file
+	// Don't panic if config.yaml is not found or error with parsing. Use the default config values instead
+	if err != nil {
+		fmt.Printf("[Config] :: Failed to parse config file from path %q, using default values: %v\n", configPath, err)
+	}
+	v.OnConfigChange(func(e fsnotify.Event) {
+		c.onConfigChange()
+	})
+	v.WatchConfig()
+
+	c.v = v
+}
+
+func (c *Config) onConfigChange() {
+	defer func() {
+		if r := recover(); r != nil {
+			err := fmt.Errorf("cannot update Config Variables: %v", r)
+			fmt.Println(err)
```

Comment:
How are we going to catch this when it happens?

Reply:
Used this piece at its original form, just moved it to another file.
I wouldn't worry that much about it though, since config files don't change randomly in our managed environments. We do changes purposefully  and then monitor, expecting to verify that our changes have actually been applied. Failure to do so, might trigger someone to have a look at the logs :)

Reply:
Makes sense at least for now it shouldn't cause major issues given that we apply and check :+1: 

I wonder what we could do to improve these bits, perhaps structured logging might be useful in these cases cc @lvrach